### ISSUE TITLE
GetModificationTime comparision for `.hi` files seems imprecise

### ISSUE THREAD

**Issue by mpickering** [haskell/ghcide#591]
Consider this section of the code

https://github.com/digital-asset/ghcide/blob/master/src/Development/IDE/Core/Rules.hs#L560-L563

If you have opened a file `f` then it will be loaded into the VFS. Therefore `GetModificationTime` will return a `VFSVersion` for `f`. For the .hi file `GetModificationTime` will return a `ModificationTime`. Therefore this comparision on line 562 will always fail for open files.

In general it doesn't seem to make sense to attempt to compare `VFSVersion` with `ModificationTime`.

**Comment by pepeiborra** [comment]
What you point out is true, it does not make sense to compare timestamps for VFS files or "files of interest" in lsp speak. 

But I believe the section of code you link never gets called for files of interest, see 

https://github.com/digital-asset/ghcide/blob/0c9a0961abbeef851b4117e6408f15a6d46eb1f1/src/Development/IDE/Core/Rules.hs#L588-L592

**Comment by mpickering** [comment]
I think there is a similar issue for .hie files. https://github.com/digital-asset/ghcide/blob/master/src/Development/IDE/Core/Rules.hs#L142

I just wanted to point this out but Zubin has rewritten most of this code to fix these problems with the hiedb patches.

**Comment by pepeiborra** [comment]
Ah, the issue for `.hie` files does indeed look like a bug (but not the one for `.hi` files).

**Comment by pepeiborra** [comment]
Actually, the code already works as it's supposed to:
```
*Development.IDE.Core.Shake Data.Ord> comparing modificationTime (ModificationTime 1 2) (VFSVersion 1)
GT
```
But it's not entirely obvious, so I'll send a PR.

My guess is that you must have observed a bug after removing `kick`, which would indeed invalidate the reasoning.

**Comment by jneira** [comment]
Closed by 8f6eb2d
